Guardian Spirits/Parasitic Nanomachine Swarm/Scarab Hive

KorPhaeron opened the pull request:
Adds a new Guardian mob, with three different fluff variants (magic/tech/bio).

When summoned/created, the mob is stored inside the summoner, and can manifest (hide back inside)/recall itself, as well as communicate privately via verb. The summoner can speak privately with the mob as well.

The mob itself is invincible, but can't live without a summoner/host, and will be warped back to them if they try to move too far away. In addition, depending on which ability set the mob has, it will deal a percentage of damage it suffers from attacks back to its host.

There are 8 variants with various quirks/powers, such as igniting enemies on touch, healing on attack, ranged attacks, etc

Added a creation tool for them, with magic/tech/bio variants.

Needs sprites for all three mob variants, as well as for the summoning items, and ideally more interesting powers later, but I think they're finished enough for playtesting (currently adminspawn only).

UPDATE: 

You can now edit which types are selectable by the spawner, and whether the spawner is random or input.

Also fixed the fluff strings the way Remie wanted them.

imtakingabreakdontatme committed:
Spirits, nanomachines, scarabs

RemieRichards commented:
add the GODMODE status flag to them to make them REALLY invincible.

daman997 commented:
so will this be added to wiz's or something later or will it just be a badmin tool forever

RemieRichards commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,400 @@
+	icon_state = "combat_hypo"
+	theme = "bio"
+	mob_name = "Scarab Swarm"
+	use_message = "The egs begin to twitch..."
```
egs

Jalleo commented:
Sounds like this can be for wiz syndie and changeling

RemieRichards commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,400 @@
+			pickedtype = /mob/living/simple_animal/hostile/guardian/fire
+			switch (theme)
+				if("magic")
+					user << "..And draw Atmosia, bringer of cleansing fires!"
```
can you please put these strings on the mobs?
eg:
var/magic_fluff_string
var/tech_fluff_string
var/bio_fluff_string

and then you can do ```user << initial(pickedype.magic_fluff_string)``` (you'll need to replace var/pickedtype with ```var/mob/living/simple_animal/hostile/guardian/pickedtype```

KorPhaeron commented:
@RemieRichards 
Would that fuck up the way I've done their AdjustBruteLoss?

@daman997

Remains to be seen. Could potentially work as a ling item (scarabs) or traitor/RD/nuke (nanomachines), or yeah, wizard.

Depends on how strong they end up being, how in depth they end up being, etc. Right now its all very basic and (I hope) erring on underpowered.

I could make a wizard only variant of the spawner that summons stronger versions with full spells, like fireball, timestop, lightning, etc.

RemieRichards commented:
@KorPhaeron actually it wouldn't because your AdjustBruteLoss doesn't call ..() so GODMODE won't work anyway.

tl;dr your code is fine, ignore GODMODE.

Anonus commented:
IIRC godmode doesn't work properly on simple animals anyway.

RemieRichards commented:
@Anonus Phil fixed that, iirc.

Anonus commented:
There should be a way to set a spawner to spawn only a specific type instead of it being random, for, say, cult or something.

KorPhaeron commented:
@Anonus 

I could have a var/random, and it either letting you pick via input or choose randomly based on that.

Anonus commented:
There should be two pick types, one being random and one being from an editable list(so that you could, say, make one that only ever spawned the fire guardian)

imtakingabreakdontatme committed:
Update

dannno commented:
i have been waiting for this day

imtakingabreakdontatme committed:
Clarified var names

Shadowlight213 commented:
As much as I love new things being given to R&D, I think this shouldn't be made something that could be mass produced. I'd say keep it out of R&D. They have too much cool shit already.

AnturK commented:
Alright, this might sound silly but : Will moving as the guardian inside summoner makes them burst ? Because i'm pretty sure that's default action for living mobs in other living mobs.

Durkel commented:
Looks like it's time to stand proud.

TheVekter commented:
So they're Stands.

imtakingabreakdontatme committed:
More updates

Punching variant can choose battlecry, removed ventcrawl from scout
(weird interactions with incorporeal move), added an explosive act.

imtakingabreakdontatme committed:
Merge branch 'korsbizzareadventure' of https://github.com/KorPhaeron/kortg into korsbizzareadventure

imtakingabreakdontatme committed:
Another update

Fixed the bluespace one telporting APCs around, added sprites for the
ghost version, fixed naming.

imtakingabreakdontatme committed:
New Guardian Type and fixes

Bomb type that sets booby traps.

Various value tweaks, bug fix, death messages.

KorPhaeron commented:
@RemieRichards 

Since it's all self contained adminspawn only code and won't break/unbalance anything else, any chance of this getting merged tonight (if you get a chance to review the new stuff) so I can test it a couple times before I leave on Friday?

Anonus commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,526 @@
+	if(summoner)
+		if(summoner.stat == DEAD)
+			src << "<span class='danger'>Your summoner has died!</span>"
+			visible_message("<span class='danger'>The [src] dies along with it's user!</span>")
```
its

Anonus commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,526 @@
+			qdel(src)
+	else
+		src << "<span class='danger'>Your summoner has died!</span>"
+		visible_message("<span class='danger'>The [src] dies along with it's user!</span>")
```
its

imtakingabreakdontatme committed:
I can't spell

RemieRichards commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,526 @@
+	stored_obj = A
+	name = A.name
+	desc = A.desc
+	icon = A.icon
```
replace:
icon
icon_state
overlays
and I believe the pixel_X/Ys

with appearance = A.appearance

RemieRichards commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,526 @@
+	pixel_y = A.pixel_y
+	pixel_x = A.pixel_x
+	spawn(600)
+		stored_obj.loc = get_turf(src.loc)
```
```stored_obj``` might not exist anymore
```spawner``` might not exist anymore
```src``` might not exist anymore

600 ticks is 60 seconds, a lot can happen in a minute.

imtakingabreakdontatme committed:
More efficient disguising

imtakingabreakdontatme committed:
Removes nonfunctional bolding

RemieRichards commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,526 @@
+	set category = "Guardian"
+	set desc = "Choose what you shout as you punch"
+	var/input = stripped_input(src,"What do you want your battlecry to be? Max length of 6 characters.", ,"", 6)
+	src.battlecry = input
```
please check input is not an empty string first.

RemieRichards commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,526 @@
+
+/mob/living/simple_animal/hostile/guardian/ex_act(severity, target)
+	switch (severity)
+		if (1.0)
```
remove the ```.0``` it's found in other procs because it's a decompilation artefact of ```1``` --> ```1.0```

imtakingabreakdontatme committed:
Ancient code

imtakingabreakdontatme committed:
Check for null input

imtakingabreakdontatme committed:
Don't need those anyway
